sensor/common/scan/scan.go

```diff
@@ -163,57 +131,171 @@ func (s *LocalScan) enrichLocalImageFromRegistry(ctx context.Context, centralCli
 
-	image := types.ToImage(ci)
-	image.Notes = make([]storage.Image_Note, 0)
-
 	// Enrich image with metadata from one of registries.
-	reg := s.enrichImageWithMetadata(errorList, registries, image)
+	reg, pullSourceImage := s.getImageWithMetadata(ctx, errorList, namespace, srcImage)
 
 	// Perform partial scan (image analysis / identify components) via local scanner.
-	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, image)
+	scannerResp := s.fetchImageAnalysis(ctx, errorList, reg, pullSourceImage)
 
 	// Fetch signatures associated with image from registry.
-	sigs := s.fetchSignatures(ctx, errorList, reg, image)
+	sigs := s.fetchSignatures(ctx, errorList, reg, pullSourceImage)
 
 	// Send local enriched data to central to receive a fully enrich image. This includes image vulnerabilities and
 	// signature verification results.
 	centralResp, err := centralClient.EnrichLocalImageInternal(ctx, &v1.EnrichLocalImageInternalRequest{
-		ImageId:        image.GetId(),
-		ImageName:      image.GetName(),
-		Metadata:       image.GetMetadata(),
+		ImageId:        srcImage.GetId(),
+		ImageName:      srcImage.GetName(),
+		Metadata:       pullSourceImage.GetMetadata(),
 		Components:     scannerResp.GetComponents(),
 		Notes:          scannerResp.GetNotes(),
 		ImageSignature: &storage.ImageSignature{Signatures: sigs},
-		ImageNotes:     image.GetNotes(),
+		ImageNotes:     pullSourceImage.GetNotes(),
 		Error:          errorList.String(),
 		RequestId:      requestID,
 		Force:          force,
 	})
 	if err != nil {
-		log.Debugf("Unable to enrich image %q: %v", image.GetName().GetFullName(), err)
-		return nil, pkgErrors.Wrapf(err, "enriching image %q via central", image.GetName())
+		log.Debugf("Unable to enrich image %q: %v", srcImage.GetName().GetFullName(), err)
+		return nil, pkgErrors.Wrapf(err, "enriching image %q via central", srcImage.GetName())
 	}
 
 	if errorList.Empty() {
-		log.Debugf("Retrieved image enrichment results for %q with id %q", image.GetName().GetFullName(), image.GetId())
+		log.Debugf("Retrieved image enrichment results from Central for %q with id %q (%v) components", srcImage.GetName().GetFullName(), srcImage.GetId(), centralResp.GetImage().GetComponents())
 	}
 
 	return centralResp.GetImage(), errorList.ToError()
 }
 
+// getImageWithMetadata on success returns the registry used to pull metadata and an image with metadata populated.
+// The image returned may represent the source image or an image from a registry mirror.
+func (s *LocalScan) getImageWithMetadata(ctx context.Context, errorList *errorhelpers.ErrorList, namespace string, sourceImage *storage.ContainerImage) (registryTypes.ImageRegistry, *storage.Image) {
+	// Obtain the pull sources, which will include mirrors.
+	pullSources := s.getPullSources(sourceImage)
+
+	// errs are only added to errorList when attempts from all pull sources + all registries fail.
+	errs := errorhelpers.NewErrorList("")
```

Comment:
can you add some kind of message to make it clear these errors are under some umbrella? For example, maybe "getting image with metadata"

Reply:
The message of that object is not actually used or printed, its only a temporary container of errors, eventually the errors are added to another errorList which does have an appropriate message.